packages/orchestration/src/exos/time.js

```diff
@@ -0,0 +1,76 @@
+import { RelativeTimeRecordShape, TimeMath } from '@agoric/time';
+import { VowShape } from '@agoric/vow';
+import { watch, allVows } from '@agoric/vow/vat.js';
+import { makeHeapZone } from '@agoric/zone';
+import { E } from '@endo/far';
+import { M } from '@endo/patterns';
+
+/**
+ * @import {Remote} from '@agoric/internal';*
+ * @import {RelativeTimeRecord, TimerBrand, TimerService, TimestampRecord} from '@agoric/time';
+ * @import {Vow} from '@agoric/vow';
+ * @import {Zone} from '@agoric/zone';
+ */
+
+export const SECONDS_PER_MINUTE = 60n;
+export const NANOSECONDS_PER_SECOND = 1_000_000_000n;
+
+/**
+ * @param {Remote<TimerService>} timerService
+ * @param {Zone} [zone]
+ */
+
+export const makeTimestampHelper = (timerService, zone = makeHeapZone()) => {
+  /** @type {TimerBrand | undefined} */
+  let brandCache;
+  const getBrand = () => {
+    if (brandCache) return brandCache;
```

Comment:
Does this early return need to be a promise? 
```suggestion
    if (brandCache) return Promise.resolve(brandCache);
```

Reply:
Updated to `return watch(brandCache)`